pyanaconda/modules/payloads/source/source_base.py

```diff
@@ -134,3 +136,31 @@ def tear_down_with_tasks(self):
         """
         task = TearDownMountTask(self._mount_point)
         return [task]
+
+
+class RPMSourceInterface(ABC):
+    """Interface class which has to be implented by all sources used by DNF payload."""
+
+    @abstractmethod
+    def generate_repo_configuration(self):
+        """Generate RepoConfigurationData structure.
+
+        This structure will be used by DNF payload in the main process.
+
+        FIXME: This is a temporary solution. Will be removed after DNF payload logic is moved.
+        """
+        pass
+
+
+class RPMMountingSourceBase(MountingSourceBase, ABC):
+    """Enhanced MountingSourceBase class to handle code specific for RPM sources."""
+    # pylint: disable=abstract-method
+
+    def generate_repo_configuration(self):
+        """Generate RepoConfigurationData structure."""
+        data = RepoConfigurationData()
+
+        data.name = self.type.value
```

Comment:
About the repo name, shouldn't we investigate it a little further? Maybe DNF will generate its own name if none is set.

Reply:
AFAIK (and I'll check that to be sure) we are always setting name for DNF Repo. For the base repo it is `anaconda`. I've created this as a type identifier which could be handy but maybe we can remove this completely and just not set it.